### PR TITLE
Feature/dax openrtb Implementation

### PR DESCRIPTION
**Pull request by sumantkad**
PFA the test calls .
I see they have mentioned about cookieSync in the documentation, The Url provided in dax documentation doesn’t seem to support GDPR macro, so have left it as is for now

[Archive.zip](https://github.com/remixd-media/prebid-server/files/6127468/Archive.zip)
Screen Shot 2021-03-11 at 9 57 25 PM

### CODE REVIEW

File: adapters/dax/dax.go [hunk 1]
```diff
@@ -4,94 +4,59 @@ import (
...
-	requests := []*adapters.RequestData{}
+	impressions := request.Imp
+	adapterRequests := make([]*adapters.RequestData, 0, len(impressions))
+	errs := make([]error, 0, len(impressions))
 
-	errors := []error{}
-
-	for i := 0; i < numImps; i++ {
-		imp := request.Imp[i]
+	if impressions == nil || len(impressions) == 0 {
```
Comment (FranStanic) [inline adapters/dax/dax.go]: `impressions == nil` check can be removed, `len(impressions) == 0` is enough.

File: adapters/dax/dax.go [hunk 2]
```diff
@@ -100,118 +65,105 @@ func (adapter *DaxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *ad
...
+				continue
+			}
+			if len(vast.Ads) == 0 {
+				errs = append(errs, &errortypes.BadServerResponse{
+					Message: fmt.Sprintf("No Ads in VAST response"),
+				})
+				continue
+			}
+			var price float64
+			price, err = strconv.ParseFloat(vast.Ads[0].InLine.Pricing, 64)
```
Comment (FranStanic) [inline adapters/dax/dax.go:139]: This error should be checked?

**Review by FranStanic — COMMENTED** [review]
LGTM, we can merge after comments are addressed 👍

File: adapters/dax/dax.go (hunk 2 above)
Comment (sumantkad) [reply adapters/dax/dax.go:139]: Sure, What do web want to do if let’s say price is empty, I guess keep it 0
For other cases when it breaks we will throw error ?

Comment (FranStanic) [reply adapters/dax/dax.go:139]: That sounds good 👍

File: adapters/dax/dax.go (hunk 1 above)
Comment (sumantkad) [reply adapters/dax/dax.go]: good point, removed it

Comment (sumantkad) [reply adapters/dax/dax.go]: Changes are pushed

File: adapters/dax/dax.go (hunk 2 above)
Comment (sumantkad) [reply adapters/dax/dax.go:139]: Changes are pushed

**Pull request merged**